code/__DEFINES/dcs/signals.dm

```diff
@@ -201,6 +201,11 @@
 	/// When returned from `COMSIG_ATOM_SINGULARITY_TRY_MOVE`, the singularity will move to that turf
 	#define SINGULARITY_TRY_MOVE_BLOCK (1 << 0)
 
+/// from base of [/atom/proc/get_material_composition]: (breakdown_flags, /list)
+#define COMSIG_ATOM_GET_MAT_COMP	"atom_get_material_composition"
```

Comment:
Pretty sure this would be in the domain of "signals as procs" that we'd rather you just use GetElement directly for. Not super educated on this bit in particular, though.

Reply:
I would disagree, if only because it's called when something tries to retrieve materials from the atom and is responding to that. @ninjanomnom What's your take?

Reply:
It also might be slightly more extensible. I really just made it this way because it seemed like it would be easier to expand on later.

Reply:
It's kind of okay. the event is fired from one place (e.g. when get_material_composition is ran), which is an atom level proc. This is similar to Examine() which is basically the same except the proc handles some other things. However, its not really great either as this functionality is just a simple trait check that could be in the proc, and the get_material_composition proc is really just a getter/helper which means the signal isn't really that great either.

Reply:
It's kind of okay. the event is fired from one place (e.g. when get_material_composition is ran), which is an atom level proc. This is similar to Examine() which is basically the same except the proc handles some other things. However, its not really great either as this functionality is just a simple trait check that could be in the proc, and the get_material_composition proc is really just a getter/helper which means the signal isn't really that great either.